src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -202,20 +233,72 @@ DatabaseType.PostgreSQL or
             }
 
             List<IDefinitionNode> nodes = new(objectTypes.Values);
-            return (new DocumentNode(nodes.Concat(inputObjects.Values).ToImmutableList()), inputObjects);
+            return new DocumentNode(nodes);
         }
 
-        private (DocumentNode, Dictionary<string, InputObjectTypeDefinitionNode>) GenerateCosmosGraphQLObjects()
+        /// <summary>
+        /// Generates the ObjectTypeDefinitionNodes and InputObjectTypeDefinitionNodes as part of GraphQL Schema generation for cosmos db.
+        /// Each datasource in cosmos has a root file provided which is used to generate the schema. 
+        /// </summary>
+        /// <param name="dataSourceNames">Hashset of datasourceNames to generate cosmos objects.</param>
+        /// <returns></returns>
+        private DocumentNode GenerateCosmosGraphQLObjects(HashSet<string> dataSourceNames, Dictionary<string, InputObjectTypeDefinitionNode> inputObjects)
         {
-            Dictionary<string, InputObjectTypeDefinitionNode> inputObjects = new();
-            DocumentNode root = ((CosmosSqlMetadataProvider)_sqlMetadataProvider).GraphQLSchemaRoot;
+            DocumentNode? root = null;
+
+            if (dataSourceNames.Count() == 0)
+            {
+                return new DocumentNode(new List<IDefinitionNode>());
+            }
+
+            foreach (string dataSourceName in dataSourceNames)
+            {
+                ISqlMetadataProvider metadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                DocumentNode currentNode = ((CosmosSqlMetadataProvider)metadataProvider).GraphQLSchemaRoot;
```

Comment:
I see that dataSourceNames are pre-filtered by the caller, so uncaught exceptions due to a failed cast may not occur here given the caller does its due diligence. I would add a comment to the method parameter stating that the list of `dataSourceNames` MUST be pre-filtered.

Reply:
added the comment.